oauthproxy.go

```diff
@@ -270,6 +269,39 @@ func (p *OAuthProxy) setupServer(opts *options.Options) error {
 	return nil
 }
 
+func (p *OAuthProxy) buildServeMux(proxyPrefix string) {
+	r := mux.NewRouter()
+	// Everything served by the router must go through the preAuthChain first.
+	r.Use(p.preAuthChain.Then)
+
+	// Register the robots path writer
+	r.Path(robotsPath).HandlerFunc(p.pageWriter.WriteRobotsTxt)
+
+	// The authonly path should be registered separately to prevent it from getting no-cache headers
```

Comment:
Do you have historical context why this is? (might be handy to add to the comment why it is excluded). I can't readily think of why this case for subrequests to `/oauth2/auth` need to have no-cache headers stripped, so it might be a weird corner case worth commenting?

Reply:
We excluded this from the no-cache because of users using nginx caching to reduce the number of request to OAuth2 Proxy. By setting the no cache headers this caching is prevented. By allowing caching here, nginx can effectively do a single flight and return that result to multiple concurrent requests.

I used to have this set up at Pusher and had a 3 second cache. Greatly reduced the chance of double refreshes and the logout issues those incur.